src/constants.ts

```diff
@@ -11,27 +11,88 @@ export const WRITING_STYLE_SIZE_LIMIT = 10000;
 export const MAX_HEADING_CHAR_LENGTH = 50;
 export const NUMBER_OF_CHARACTERS_TO_FLUSH_TO_FILE = 200;
 
-export const CLAUDE_MODELS = [
+// MUST UPDATE `AI_PROVIDERS` IN ai.ts WHEN NEW PROVIDER ADDED
+export const AI_MODELS_UI = [
   {
     name: "Claude 3 Opus",
+    provider: "anthropic",
     model: "claude-3-opus-20240229",
     smarterDescription: "This is the ferrari. Expensive but so good.",
     pageDescription: "Smartest - Use for expensive but awesome results!"
   },
   {
     name: "Claude 3 Sonnet",
+    provider: "anthropic",
     model: "claude-3-sonnet-20240229",
     smarterDescription: "Perfectly fine!",
     pageDescription: "Middle child - still kind of expensive"
   },
   {
     name: "Claude 3 Haiku",
+    provider: "anthropic",
     model: "claude-3-haiku-20240307",
     smarterDescription: "Cheapest, not preferred for this stage",
     pageDescription: "Fast and cheap - get what we pay for"
+  },
+  {
+    name: "OpenAI GPT-3.5 Turbo (1106)",
+    provider: "openai",
+    model: "gpt-3.5-turbo-1106",
+    tokenCountingModel: "gpt-3.5-turbo", // required bc OpenAI token counting is frustrating
+    smarterDescription:
+      "Pretty cheap, good at JSON, bit dumb for the management stuff",
+    pageDescription: "Fast, cheap, lower rate limit than Claude"
+  },
+  {
+    name: "OpenAI GPT-4 Turbo",
+    provider: "openai",
+    model: "gpt-4-turbo-2024-04-09",
+    tokenCountingModel: "gpt-4", // required bc OpenAI token counting is frustrating
+    smarterDescription: "Bit expensive but smart",
+    pageDescription:
+      "Good for high quality docs when you need to make a lot and are willing to pay"
   }
 ] as const;
 
+// MUST UPDATE `AI_PROVIDERS` IN ai.ts WHEN NEW PROVIDER ADDED
+export const AI_MODELS_INFO: Record<
+  string,
+  {
+    provider: "anthropic" | "openai";
+    tokenCountingModel?: string;
+    inputTokensPerM: number;
+    outputTokensPerM;
+  }
+> = {
+  "claude-3-opus-20240229": {
+    provider: "anthropic",
+    inputTokensPerM: 15,
+    outputTokensPerM: 75
+  },
+  "claude-3-sonnet-20240229": {
+    provider: "anthropic",
+    inputTokensPerM: 3,
+    outputTokensPerM: 15
+  },
+  "claude-3-haiku-20240307": {
+    provider: "anthropic",
+    inputTokensPerM: 0.25,
+    outputTokensPerM: 1.25
+  },
+  "gpt-4-turbo-2024-04-09": {
+    provider: "openai",
+    tokenCountingModel: "gpt-4", // required bc OpenAI token counting is frustrating
+    inputTokensPerM: 10,
```

Comment:
Jeez that's EX

Reply:
that's the cheaper one

Reply:
but pls do double check my pricing it would be embarassing to get that wrong

Reply:
dbl checked